### PR TITLE
NIFI-957 Added the possibility to use DefaultSchedule annotation in r…

### PR DESCRIPTION
**Pull request by pvillard31**
…eporting tasks

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [X] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by olegz** [comment]
Reviewing. . .
@pvillard31 could you please address the merge conflict, looks trivial

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/reporting/AbstractReportingTaskNode.java [hunk 1]
```diff
@@ -72,6 +77,27 @@ public AbstractReportingTaskNode(final ReportingTask reportingTask, final String
         this.reportingTask = reportingTask;
         this.processScheduler = processScheduler;
         this.serviceLookup = controllerServiceProvider;
+
+        final Class<?> reportingClass = reportingTask.getClass();
+
+        try {
+            if (reportingClass.isAnnotationPresent(DefaultSchedule.class)) {
+                DefaultSchedule dsc = reportingClass.getAnnotation(DefaultSchedule.class);
```
Comment (olegz) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/reporting/AbstractReportingTaskNode.java]: I am wondering if we should instead use Spring's AnnotationUtils and its discovery operations there. Basically the question is if you *only* looking for annotation in the specific class _reportingTask.getClass()_, then the above is OK i guess, but I am afraid that if impl of the ReportingTask has a more complex hierarchy and annotated operation is in some super class, then the above will not work. WDYT?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/reporting/AbstractReportingTaskNode.java [hunk 2]
```diff
@@ -72,6 +77,27 @@ public AbstractReportingTaskNode(final ReportingTask reportingTask, final String
...
+                }
+                try {
+                    this.setSchedulingPeriod(dsc.period());
+                } catch (Throwable ex) {
+                    this.setSchedulingStrategy(SchedulingStrategy.TIMER_DRIVEN);
+                    LOG.error(String.format("Error while setting scheduling period from DefaultSchedule annotation: %s", ex.getMessage()), ex);
+                }
+            }
+        } catch (Throwable ex) {
+            LOG.error(String.format("Error while setting default schedule from DefaultSchedule annotation: %s",ex.getMessage()),ex);
```
Comment (olegz) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/reporting/AbstractReportingTaskNode.java]: Wondering based on the code above and internal try/catch is it even possible to get here?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/reporting/AbstractReportingTaskNode.java (hunk 1 above)
Comment (pvillard31) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/reporting/AbstractReportingTaskNode.java]: You're right, I updated the PR.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/reporting/AbstractReportingTaskNode.java (hunk 2 above)
Comment (pvillard31) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/reporting/AbstractReportingTaskNode.java]: Correct, removed unnecessary try/catch.

**Comment by pvillard31** [comment]
Hey @olegz, thanks for reviewing this PR. I addressed your comments, let me know if there is something else.

**Pull request closed without merging**